samples/subsys/usb/uvc/README.rst

```diff
@@ -0,0 +1,143 @@
+.. zephyr:code-sample:: uvc
+   :name: USB Video sample
+   :relevant-api: usbd_api video_interface
+
+   USB Video sample sending frames over USB.
+
+Overview
+********
+
+This sample demonstrates how to use an USB Video Class instance to
+send video data over USB.
+
+Upon connection, a video interface would show-up in the operating system,
+using the same protocol as most webcams, and be detected as such.
+
+Any software application to read and process video would then be able
+to access the stream.
+
+Requirements
+************
+
+USB is the only requirement for this sensor, as an emulated image
+sensor and MIPI receiver is used to provide a test pattern from
+software.
+
+The USB descriptor configuration is extracted from these drivers, and
+the user does not need to provide them.
+
+Building and Running
+********************
+
+Build the sample application and flash the resulting binaries.
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/subsys/usb/uvc
+   :board: nrf52840dongle
+   :goals: build flash
+   :compact:
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/subsys/usb/uvc
+   :board: mini_stm32h734
+   :goals: build flash
+   :compact:
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/subsys/usb/uvc
```

Comment:
Have you tried with the Arduino Nicla Vision?

Reply:
Not yet! But I did test on one board I did happen to have with me: `stm32h743`.
I scheduled a Nicla in my order list for next batch.

Reply:
Hi @josuah, if you want any help testing, feel free to let me know. We're doing some video and streaming stuff with a few other boards (the main one is still an h7 but slightly different), and we'd be happy to help with these efforts (video, usb, etc, improvements in Zephyr) but it looks like you're much more experienced at it. Thanks for your efforts! Andrew

Reply:
Greatly appreciated @andrewnyland! I do my best to pass the wisdom from Linux, V4L2, libcamera, and [image processing gurus](https://github.com/zephyrproject-rtos/zephyr/blob/main/MAINTAINERS.yml#L2127) to Zephyr RTOS.

For the topic of UVC in particular, testing this fork and reporting issues like bugs, usability fix, limitations, etc. does help with moving things forward, and be hopefully be useful to yourself.

There is also this thread to discuss and troubleshoot if you wish:
- https://github.com/zephyrproject-rtos/zephyr/discussions/63421